fsck/fsck.c

```diff
@@ -215,6 +226,38 @@ static int check_clus_chain(struct exfat_de_iter *de_iter,
 	}
 
 	return 0;
+allocate_cluster:
+	exfat_de_iter_get_dirty(de_iter, 1, &stream_de);
+	err = exfat_find_free_cluster(exfat, exfat->start_clu, &new_clus);
+
+	if (err) {
+		exfat->start_clu = EXFAT_FIRST_CLUSTER;
+		exfat_err("failed to find a free cluster\n");
+		return -ENOSPC;
+	}
+	exfat->start_clu = new_clus;
+
+	if (exfat_set_fat(exfat, new_clus, EXFAT_EOF_CLUSTER))
+		return -EIO;
+
+	/* zero out the new cluster */
+	if (exfat_write(exfat->blk_dev->dev_fd, exfat->zero_cluster,
+			exfat->clus_size, exfat_c2o(exfat, new_clus)) !=
+			(ssize_t)exfat->clus_size) {
+		exfat_err("failed to fill new cluster with zeroes\n");
+		return -EIO;
+	}
+	exfat_info("final cluster : %#x \n", new_clus);
```

Comment:
This is needed ? If not, can I remove it ?

Reply:
Sure. You can remove it.